pages/robot/tabs/components/index.php

```diff
@@ -664,31 +549,47 @@ function render_components(data) {
                 })
 
                 // looking for test records
-                $.ajax({
-                    url: window.location.href,
-                    type: "POST",
-                    data: {test_id_read: id_str_name},
-                    success: function(response) {
-                        let [datetime, passed] = parse_db_record_response(response, id_str_name);
-                        update_style_based_on_records(id_str_name, datetime, passed);
+                smartAPI(
+                    "data",
+                    "get",
+                    {
+                        arguments: {
+                            database: "<?php echo $HW_TEST_DB_NAME ?>",
+                            key: id_str_name
+                        },
+                        quiet: true,
+                        on_success: function (data) {
+                            let [datetime, passed] = parse_db_record_response(data);
+                            update_style_based_on_records(id_str_name, datetime, passed);
+                        },
                     }
-                });
+                )
+
+                let write_to_db = function(passed) {
+                    smartAPI(
+                        "data",
+                        "set",
+                        {
+                            arguments: {
+                                database: "<?php echo $HW_TEST_DB_NAME ?>",
+                                key: id_str_name,
+                                value: JSON.stringify({
+                                    passed: passed,
+                                    datetime: Date.now()
+                                })
+                            },
+                            quiet: true,
+                            reload: true
+                        }
+                    )
+                }
```

Comment:
The folder `hardware_test_result` gets created in `/user-data/databases/data` (in the **dashboard** container), but no files appear, when trying to mark "Success" nor "Problem". Also, probably for the same reason, no status were reflected once the status is marked (neither right away nor after refresh)